react/features/invite/middleware.any.js

```diff
@@ -15,9 +42,64 @@ const logger = require('jitsi-meet-logger').getLogger(__filename);
  */
 // eslint-disable-next-line no-unused-vars
 MiddlewareRegistry.register(store => next => action => {
+    let oldParticipantPresence;
+
+    if (action.type === PARTICIPANT_UPDATED) {
+        oldParticipantPresence
+            = _getParticipantPresence(store.getState(), action.participant.id);
+    }
+
     const result = next(action);
 
     switch (action.type) {
+    case APP_WILL_MOUNT:
+        store.dispatch(
+            registerSound(
+                OUTGOING_CALL_START_SOUND_ID,
+                OUTGOING_CALL_START_FILE));
+
+        store.dispatch(
+            registerSound(
+                OUTGOING_CALL_RINGING_SOUND_ID,
+                OUTGOING_CALL_RINGING_FILE,
+                { loop: true }));
+        break;
+
+    case APP_WILL_UNMOUNT:
+        store.dispatch(unregisterSound(OUTGOING_CALL_START_SOUND_ID));
+        store.dispatch(unregisterSound(OUTGOING_CALL_RINGING_SOUND_ID));
+        break;
+
+    case PARTICIPANT_LEFT:
+    case PARTICIPANT_UPDATED: {
+        const newParticipantPresence
+            = _getParticipantPresence(store.getState(), action.participant.id);
+
+        if (oldParticipantPresence === newParticipantPresence) {
```

Comment:
"oldParticipantPresence" is only defined if the action was PARTICIPANT_UPDATED. I'm not fully familiar with the logic, so just asking: is it enough to have value then?

Reply:
Are you concerned for the PARTICIPANT_LEFT action?
I'll add PARTICIPANT_LEFT in the `if`!

Reply:
Oh actually I think we are never firing PARTICIPANT_LEFT event for those kind of participants. 
I have to figure out how to detect removal of the poltergeist.

Thanks for noticing this! Good catch!